presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestCachingHiveMetastore.java

```diff
@@ -171,22 +172,55 @@ public void testGetPartitionNamesByParts()
         assertEquals(mockClient.getAccessCount(), 2);
     }
 
+    private static class MockPartitionMutator
+            implements PartitionMutator
+    {
+        public enum VersionType
+        {
+            CONSTANT,
+            INCREASING,
+            DECREASING
+        }
+
+        private final VersionType versionType;
+        private int version = 1000;
+
+        public MockPartitionMutator(VersionType versionType)
+        {
+            this.versionType = versionType;
+        }
+
+        @Override
+        public void mutate(Builder builder, org.apache.hadoop.hive.metastore.api.Partition partition)
+        {
+            switch (versionType) {
+                case INCREASING:
+                    builder.setPartitionVersion(++version);
+                    break;
+                case DECREASING:
+                    builder.setPartitionVersion(--version);
+                    break;
+                default:
```

Comment:
Let's explicitly make a case for `CONSTANT` (just break should be good enough) and throw for `default`.